javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -218,6 +290,114 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
     }
 
     /**
+     * Creates a {@code TreeMap} of the given list of key-value pairs.
+     *
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K k1, V v1, K k2, V v2) {
+        return TreeMap.<K, V> empty(keyComparator).put(k1, v1).put(k2, v2);
+    }
+
+    /**
+     * Creates a {@code TreeMap} of the given list of key-value pairs.
+     *
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K k1, V v1, K k2, V v2, K k3, V v3) {
+        return TreeMap.<K, V> empty(keyComparator).put(k1, v1).put(k2, v2).put(k3, v3);
+    }
+
+    /**
+     * Creates a {@code TreeMap} of the given list of key-value pairs.
+     *
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4) {
+        return TreeMap.<K, V> empty(keyComparator).put(k1, v1).put(k2, v2).put(k3, v3).put(k4, v4);
+    }
+
+    /**
+     * Creates a {@code TreeMap} of the given list of key-value pairs.
+     *
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4, K k5, V v5) {
+        return TreeMap.<K, V> empty(keyComparator).put(k1, v1).put(k2, v2).put(k3, v3).put(k4, v4).put(k5, v5);
+    }
+
+    /**
+     * Creates a {@code TreeMap} of the given list of key-value pairs.
+     *
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @param <K> The key type
+     * @param <V> The value type
+     * @return A new Map containing the given entries
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4, K k5, V v5, K k6, V v6) {
```

Comment:
in other places the params have JavaDoc also (I don't think they have value, though, but it seems inconsistent to me)